php/src/Docker/DockerActionManager.php

```diff
@@ -83,21 +79,21 @@ public function GetContainerRestartingState(Container $container) : IContainerSt
             $response = $this->guzzleClient->get($url);
         } catch (RequestException $e) {
             if ($e->getCode() === 404) {
-                return new ImageDoesNotExistState();
+                return IContainerState::ImageDoesNotExist;
             }
             throw $e;
         }
 
         $responseBody = json_decode((string)$response->getBody(), true);
 
         if ($responseBody['State']['Restarting'] === true) {
-            return new RestartingState();
+            return IContainerState::Restarting;
         } else {
-            return new NotRestartingState();
+            return IContainerState::NotRestarting;
         }
     }
 
-    public function GetContainerUpdateState(Container $container) : IContainerState
+    public function GetContainerUpdateState(Container $container) : bool
```

Comment:
I liked the old, more descriptive version better. A boolean return type could be obscure and easy to misinterpret.

Perhaps you could add another enum or add the two extra states to the existing ones?